src/siminterface/simulator.py

```diff
@@ -78,6 +78,12 @@ def apply(self, actions: SimulatorAction):
         # Get the new placement from the action passed by the RL agent
         # Modify and set the placement parameter of the instantiated simulator object.
         self.simulator.params.sf_placement = actions.placement
+        # Update which sf is available at which node
+        for node_id, placed_sf_list in actions.placement.items():
+            available_sf = {}
+            for sf in placed_sf_list:
+                available_sf[sf] = self.simulator.params.network.nodes[node_id]['available_sf'].get(sf, {'load': 0.0})
```

Comment:
What does this do exactly?

Reply:
Each node stores a list of SFs that where placed and there current load as a node attribute. When changing the placement the list has to be updated.

Reply:
So the `.get(sf, {'load': 0.0})` gets the current load of the SF at the node and defaults to 0 if the SF wasn't available?